recipes-ni/niconfigpersistentlogs/files/niconfigpersistentlogs

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+set -uo pipefail
+
+error_and_die() {
+set +u
+    if [ -n "$1" ]; then
+        echo >&2 "$1"
+    fi
+set -u
+    echo >&2 "ERROR"
+    exit 1
+}
+
+FILE_PATH_VOLATILES_CONFIG="/etc/default/volatiles/00_core"
+PERSISTENT_DIR_VAR_LOG_LINE="d root root 0755 /var/log none"
+VOLATILE_SYMLINK_VAR_LOG_LINE="l root root 0755 /var/log /var/volatile/log"
+
+ENABLE_PERSISTENT_LOGS_INI_VAL=$(/usr/local/natinst/bin/nirtcfg --get section=SystemSettings,token=PersistentLogs.enabled,value="false" | tr "[:upper:]" "[:lower:]")
+
+case $ENABLE_PERSISTENT_LOGS_INI_VAL in
+    true)
+        NEW_VAR_LOG_LINE="$PERSISTENT_DIR_VAR_LOG_LINE"
+        OLD_VAR_LOG_LINE="$VOLATILE_SYMLINK_VAR_LOG_LINE"
+        SUCCESS_MESSAGE="Persistent logs enabled."
+        RM_VAR_LOG=true
+        ;;
+    false)
+        NEW_VAR_LOG_LINE="$VOLATILE_SYMLINK_VAR_LOG_LINE"
+        OLD_VAR_LOG_LINE="$PERSISTENT_DIR_VAR_LOG_LINE"
+        SUCCESS_MESSAGE="Persistent logs disabled."
+        RM_VAR_LOG=false
+        ;;
+    *)
+        error_and_die "Unexpected value for PersistentLogs.enabled in /etc/natinst/share/ni-rt.ini. Expected True or False."
+        ;;
+esac
+
+# This grep command looks for the exact new line.
+grep "^$NEW_VAR_LOG_LINE$" "$FILE_PATH_VOLATILES_CONFIG" > /dev/null
+EXIT_CODE="$?"
+
+if [ "$EXIT_CODE" = 0 ]; then
+set +u
+    # New line is already present, do not modify file.
+    [ "${VERBOSE}" != "no" ] && echo "No changes needed to $FILE_PATH_VOLATILES_CONFIG."
+set -u
+elif [ "$EXIT_CODE" = 1 ]; then
+    # This grep command looks for the exact old line, which we want to replace with the new line.
+    grep "^$OLD_VAR_LOG_LINE$" "$FILE_PATH_VOLATILES_CONFIG" > /dev/null
+    EXIT_CODE="$?"
+
+    if [ "$EXIT_CODE" = 0 ]; then
+        # This sed command replaces the old line with the new line.
+        if ! sed "s|^$OLD_VAR_LOG_LINE$|$NEW_VAR_LOG_LINE|" -i "$FILE_PATH_VOLATILES_CONFIG"
+        then
+        error_and_die
+        fi
+    elif [ "$EXIT_CODE" = 1 ]; then
+        error_and_die "Could not locate expected /var/log entry in $FILE_PATH_VOLATILES_CONFIG."
+    else
+        error_and_die
+    fi
+else
+    error_and_die
+fi
```

Comment:
Rather than `sed`ing a file which is owned by another package, what if we just removed the `/var/log` declaration from `00_core` and used a new runparts file in the `/etc/default/volatiles` directory which is wholely owned by this recipe? Then this script's logic is just to swap out the contents of its runparts file wholesale.

Reply:
We had some offline discussion and decided to leave the current approach intact. Moving this line out to a separate runparts file had its own complications:

1. There are ordering dependencies with other entries in 00_core, such as those defining /var/volatile/log and /var/log/wtmp. So we would have to move at least one of those out as well. There's also a potential issue that other upstream changes to 00_core could add additional dependencies over time that we'd have to deal with.

2. By moving the /var/log line out of 00_core and into a separate runparts file in the ni-configpersistentlogs package, we would be making ni-configpersistentlogs a required package to have /var/log defined on a system. We don't really want ni-configpersistentlogs to be required. Specifically, we don't want it to be part of a safemode image where the entire rootfs is RAM-based and the whole idea of persistent logs doesn't really make sense.